operate/webapp/src/main/java/io/camunda/operate/webapp/backup/BackupRepository.java

```diff
@@ -23,4 +30,25 @@ public interface BackupRepository {
 
   void executeSnapshotting(
       BackupService.SnapshotRequest snapshotRequest, Runnable onSuccess, Runnable onFailure);
+
+  default boolean isIncompleteCheckTimedOut(
+      final OperateProperties operateProperties,
+      final long startTimeInMilliseconds,
+      final long endTimeInMilliseconds) {
+    try {
+      final var now = Instant.now().toEpochMilli();
+      final var timeout = operateProperties.getBackup().getIncompleteCheckTimeoutInSeconds();
+      // if is no end time available we just check start time with now
+      if (now - startTimeInMilliseconds > timeout && endTimeInMilliseconds <= 0) {
+        return true;
+      }
+      final long timeOutInSeconds =
+          operateProperties.getBackup().getIncompleteCheckTimeoutInSeconds();
+      return (endTimeInMilliseconds - startTimeInMilliseconds) > timeOutInSeconds;
```

Comment:
Are we comparing milliseconds with seconds ? Shouldn't it be > (timeOutInSeconds * 1000) ?

Reply:
Good catch. Thanks.